compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -124,24 +124,30 @@ object SyntaxHighlighting {
         }
       }
 
-      val parser = new Parser(source)
-      val trees = parser.blockStatSeq()
-      TreeHighlighter.highlight(trees)
-
-      val highlighted = new StringBuilder()
-
-      for (idx <- colorAt.indices) {
-        val prev = if (idx == 0) NoColor else colorAt(idx - 1)
-        val curr = colorAt(idx)
-        if (curr != prev)
-          highlighted.append(curr)
-        highlighted.append(in(idx))
-      }
+      try
+        val parser = new Parser(source)
+        val trees = parser.blockStatSeq()
+        TreeHighlighter.highlight(trees)
+
+
+        val highlighted = new StringBuilder()
 
-      if (colorAt.last != NoColor)
-        highlighted.append(NoColor)
+        for (idx <- colorAt.indices) {
+          val prev = if (idx == 0) NoColor else colorAt(idx - 1)
+          val curr = colorAt(idx)
+          if (curr != prev)
+            highlighted.append(curr)
+          highlighted.append(in(idx))
+        }
+
+        if (colorAt.last != NoColor)
+          highlighted.append(NoColor)
 
-      highlighted.toString
+        highlighted.toString
+      catch
+        case(e): StackOverflowError =>
```

Comment:
This is a very strange style. We typically write such clauses like this:
```suggestion
        case e: StackOverflowError =>
```